rflx/generator/generator.py

```diff
@@ -3253,9 +3252,17 @@ def __create_valid_context_function(
                         String('"Buffer" is not modified, could be of access constant type'),
                     ],
                 ),
+                Pragma(
+                    "Warnings",
+                    [
+                        Variable("Off"),
+                        String("postcondition does not mention function result"),
+                    ],
+                ),
                 ExpressionFunctionDeclaration(
                     specification,
                     common.context_predicate(message, composite_fields, self.__prefix),
+                    [Postcondition(TRUE)],
```

Comment:
I suppose that is just a temporary workaround? A workaround should be tracked by adding a reference to a RecordFlux issue or a [workaround](https://github.com/Componolit/Workarounds/issues) issue, so that we know why we added it and ensure that it is removed in the future.

Reply:
Not necessarily, adding such dummy post conditions is currently the only way to avoid inlining in certain contexts for proof, and it's not clear this will change in SPARK. Shall I add a comment instead?

Reply:
Yes, a comment explaining that would be nice.

Reply:
done

Reply:
> Not necessarily, adding such dummy post conditions is currently the only way to avoid inlining in certain contexts for proof, and it's not clear this will change in SPARK.

The need for suppressing the warning makes that quite cumbersome. I think it would be an improvement if there was a way to annotate a function without getting a warning as a side effect. Maybe you could consider that as an enhancement for SPARK.

Reply:
Yes, we discussed this in the SPARK team and opened V209-006. I expect this to be implemented soon.